src/main/java/org/springframework/data/couchbase/core/convert/AbstractCouchbaseConverter.java

```diff
@@ -74,6 +73,7 @@ public ConversionService getConversionService() {
 	 */
 	public void setCustomConversions(final CustomConversions conversions) {
 		this.conversions = conversions;
+		afterPropertiesSet(); // register the conversions with the conversion service
```

Comment:
Please refrain from calling lifecycle methods in setters. Calling `afterPropertiesSet` is a responsibility of the bean container.

Reply:
I can fix that by instead calling conversions.registerConvertersIn(conversionService) directly here.

The underlying issue is that 
(a) setCustomConversions() doesn't do everything needed to setCustomConversions(). It only does this.conversions = conversions and that isn't sufficient. It is also necessary to conversions.registerConvertersIn(conversionService) - otherwise the conversionService is not aware of the conversions. I made the call to afterPropertiesSet() in the event that in the future additional actions were added.
(b) the mapping converter was created outside the Bean container - in #1141 - although that should be unnecessary by  creating the template for configureRepositoryOperationsMapping using the converter (MappingCouchbaseConverter) (baseMapping.getDefault().getConverter()) . 

Somewhat related - Intellij complains when an @Bean is called directly, but sometimes it is necessary - when the value of an @Bean method is being used in another public @Bean, and the signature of that public @Bean cannot (should not?) be changed.  This is the situation with  On the other hand - if there was a guarantee that no applications were calling the @Bean method directly, then changing the signature should not be a problem. This issue exists in three bean methods in AbstractCouchbaseConfiguration.   The "solution" has been to make @Bean method call - the problem with this is that the result of that call is not the actual @Bean, it's a different object. (I suppose that if the result object was cached and used on subsequent calls, then it could be the same object).

couchbaseObjectMapper() is an @Bean method, it is now required by the couchbaseClusterEnvironment() @Bean, but was not/is not a method argument
```
	@Bean(destroyMethod = "shutdown")
	public ClusterEnvironment couchbaseClusterEnvironment() 
	   ... 
	   builder.jsonSerializer(JacksonJsonSerializer.create(couchbaseObjectMapper()));
```

This couchbaseTemplate() method was previously the @Bean, but has been superseded with a method with an additional argument.   
```
	public CouchbaseTemplate couchbaseTemplate(CouchbaseClientFactory couchbaseClientFactory,
			MappingCouchbaseConverter mappingCouchbaseConverter) {
		return couchbaseTemplate(couchbaseClientFactory, mappingCouchbaseConverter, new JacksonTranslationService());
	}
```

Reply:
> setCustomConversions() doesn't do everything needed to setCustomConversions(). It only doe

That's by intent. The bean configuration phase consists from calling all setters first and then initialize the bean with `afterPropertiesSet`. That being said, calling `registerConvertersIn(…)` in the setter isn't better.

>  the mapping converter was created outside the Bean container 

It is then the responsibility of the creator to initialize that bean.

Reply:
ok. I'll just remove the call and not replace it with anything.